development/image-url-helper/pkg/common/valuefile.go

```diff
@@ -17,22 +17,44 @@ type ValueFile struct {
 	Global GlobalKey `yaml:"global,omitempty"`
 }
 
-func AppendImagesToMap(parsedFile ValueFile, images, testImages ImageMap, component string, components ImageToComponents) {
+func AppendImagesToMap(parsedFile ValueFile, images, testImages ComponentImageMap, component string) {
 	for _, image := range parsedFile.Global.Images {
 		// add registry info directly into the image struct
 		if image.ContainerRegistryURL == "" {
 			image.ContainerRegistryURL = parsedFile.Global.ContainerRegistry.Path
 		}
-		images[image.FullImageURL()] = image
 
-		components[image.FullImageURL()] = append(components[image.FullImageURL()], component)
+		if _, ok := images[image.FullImageURL()]; ok {
+			componentImage := images[image.FullImageURL()]
+			componentImage.Components[component] = true
+			images[image.FullImageURL()] = componentImage
```

Comment:
Can you work directly on images `ComponentImageMap`?
```suggestion
                        images[image.FullImageURL()].Components[component] = true
```